ssl/ssl_lib.c

```diff
@@ -4516,6 +4516,37 @@ size_t SSL_early_get0_compression_methods(SSL *s, const unsigned char **out)
     return s->clienthello->compressions_len;
 }
 
+int SSL_early_get1_extensions_present(SSL *s, int **out, size_t *outlen)
+{
+    RAW_EXTENSION *ext;
+    size_t num = 0, i;
+
+    if (s->clienthello == NULL || out == NULL || outlen == NULL)
+        return 0;
+    for (i = 0; i < s->clienthello->pre_proc_exts_len; i++) {
+        ext = s->clienthello->pre_proc_exts + i;
+        if (ext->present)
+            num++;
+    }
+    *out = OPENSSL_malloc(sizeof(**out) * num);
```

Comment:
I'd argue in favour of not writing to `*out` in case of error. In other words pointer returned by malloc should be saved to local variable and `*out` should be assigned along with `*outlen`. Rationale is that it will give sanitizer or its equivalent chance to catch uninitialized variable in case caller misbehaves.